zapcore/buffered_write_syncer_bench_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zapcore
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkBufferedWriteSyncer(b *testing.B) {
+	b.Run("write file with buffer", func(b *testing.B) {
+		file, err := ioutil.TempFile("", "log")
+		require.NoError(b, err)
+		defer file.Close()
```

Comment:
I think this can return an error which we need to catch. Also in example in https://golang.org/pkg/io/ioutil/#TempFile i dont think they manually close the file.

Reply:
Which example are you talking about? Both examples linked there close the files at the end; they just don't use defer.
ioutil.TempFile returns an open file so we always need to close it.

RE: error: yeah, we can/should handle that. We can probably combine the two defers into a single one with two NoErrors:

```
defer func() {
  assert.NoError(t, file.Close())
  assert.NoError(t, os.Remove(file.Name())
}()
```

Reply:
Question is whether a separate explicit `file.Close` is needed or `os.Remove(file.Name())` is necessary. When @moisesvega experimented, it looked like Removing will close the file for you so a separate Close statement may be redundant and unnecessary.

Reply:
I would think that from a hygiene perspective, we'd always want to close the file descriptor once we were done with it regardless of what we were doing with the file afterwards. We shouldn't leave the file descriptor open when we get around to deleting the file; that just seems icky.

Reply:
The file descriptor does not close automatically after removing the file. If we don't close the file descriptor we would rely on the Garbage Collector to close it. This could lead to a file descriptor leak. Best practice here is run `file.Close` then `os.Remove`.

I will add the asserts inside of a defer function just to make sure that `close` and `remove` run without errors.

Reply:
thanks!